### PR TITLE
Fix dialyzer warnings

### PR DESCRIPTION
**Pull request by joaothallis**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -34,12 +34,42 @@ jobs:
           otp-version: ${{matrix.otp}}
           elixir-version: ${{matrix.elixir}}
 
+      - name: Restore dependencies cache
```
Comment (patrykwozinski) [inline .github/workflows/ci.yml:37]: Good idea!

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -34,12 +34,42 @@ jobs:
...
+        if: steps.plt-cache.outputs.cache-hit != 'true'
+        run: |
+          mkdir -p priv/plts
+          mix dialyzer --plt
+
+      - name: Run dialyzer
+        run: mix dialyzer --no-check
+
+      - run: mix test --warnings-as-errors
+        if: matrix.elixir == '1.12.1'
```
Comment (patrykwozinski) [inline .github/workflows/ci.yml:72]: Why it's only this version?

File: lib/strong_migrations.ex [hunk 3]
```diff
@@ -22,7 +22,7 @@ defmodule StrongMigrations do
       :safe
 
   """
-  @spec analyze([migration_path()]) :: :safe | :unsafe
+  @spec analyze([migration_path()]) :: :safe | {:unsafe, [reason]}
```
Comment (patrykwozinski) [inline lib/strong_migrations.ex:25]: Ah, good catch - thank you!

**Review by patrykwozinski — APPROVED** [review]
Thank you a lot for this pull request! :) I’m happy to see it.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (joaothallis) [reply .github/workflows/ci.yml:72]: `mix test` with `--warnings-as-errors` only works with Elixir 1.12+
I tried with `matrix.elixir` >= 1.12 but it does not work

Comment (patrykwozinski) [reply .github/workflows/ci.yml:72]: Ah, that's the case. Sure - anyway thanks! :)

**Pull request merged**